packages/expo-asset/src/AssetSourceResolver.web.ts

```diff
@@ -37,7 +37,7 @@ function getScaledAssetPath(asset): string {
   const scale = AssetSourceResolver.pickScale(asset.scales, getScale());
   const scaleSuffix = scale === 1 ? '' : '@' + scale + 'x';
   const assetDir = getBasePath(asset);
-  return assetDir + '/' + asset.name + scaleSuffix + '.' + asset.type;
+  return assetDir + '/' + asset.name + scaleSuffix + (asset.type ? `.${asset.type}` : '');
```

Comment:
This is needed to support Snack assets served from S3 (which don't have an extension)

Reply:
so a trailing `.` crashes it? Or is the issue that we are trying to deal with some strange handling of `null`?